src/utils/renderNamePattern.ts

```diff
@@ -1,13 +1,15 @@
 import { errFailedValidation, error } from './error';
 import { extname } from './path';
 import { isPlainPathFragment } from './relativeId';
+import { sanitizeFileName as defaultSanitizeFileName } from './sanitizeFileName';
 
 export function renderNamePattern(
 	pattern: string,
 	patternName: string,
-	replacements: { [name: string]: () => string }
+	replacements: { [name: string]: () => string },
+	sanitizeFileName = defaultSanitizeFileName
```

Comment:
I would not make this option optional as this would mean that file names are inconsistent: The option should be applied everywhere.